src/data_provider/src/UtilsWrapperUnix.cpp

```diff
@@ -0,0 +1,40 @@
+/*
+ * Wazuh SysInfo
+ * Copyright (C) 2015-2021, Wazuh Inc.
+ * December 17, 2021.
+ *
+ * This program is free software; you can redistribute it
+ * and/or modify it under the terms of the GNU General Public
+ * License (version 2) as published by the FSF - Free Software
+ * Foundation.
+ */
+
+#include <cerrno>
+#include <cstring>
+#include <system_error>
+
+#include "UtilsWrapperUnix.hpp"
+
+int UtilsWrapperUnix::createSocket(int domain, int type, int protocol)
+{
+    auto fd { socket(domain, type, protocol) };
+
+    if (-1 == fd)
+    {
+        throw std::system_error{-1, std::system_category(), "Cannot connect to local socket."};
```

Comment:
It'd be more useful if the used the `errno` error code instead of -1, which only indicates that something went wrong, while the value of `errno` can tell us why the function failed.
Maybe we could even do something like this using [strerror()](https://man7.org/linux/man-pages/man3/strerror.3.html):
```
throw std::system_error{errno, std::system_category(), strerror(errno)};
```